### PR TITLE
Lesson 5-7 part 1: Driver

### PR DESCRIPTION
**Pull request by vizhyk**
I didn't describe read function  for proc and write for sys and it's not final version, but please check my code . Maybe You will find some misstakes.

### CODE REVIEW

File: chtest.c [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+
+
+static int dev_release(struct inode *inodep, struct file *filep){
+   printk(KERN_INFO "chtest: Device successfully closed\n");
+   return 0;
+}
+ 
+module_init(chtest_init);
+module_exit(chtest_exit);
```
Comment (galkinletter) [inline chtest.c:233]: Please add empty line at the end of file

**Review by galkinletter — CHANGES_REQUESTED** [review]
Need fix code style

Makefile missing

**Comment by galkinletter** [comment]
Please push makefile also

**Comment by vizhyk** [comment]
@galkinletter , Add Makefile

File: Makefile [hunk 2]
```diff
@@ -8,3 +8,4 @@ Second.o:Second.c
 	$(CC) $(Opt) -c ./Second.c -o ./Second.o
```
Comment (galkinletter) [inline Makefile]: Makefile need for chtest.c

**Review by galkinletter — CHANGES_REQUESTED** [review]
Need fix makefile

**Review by galkinletter — CHANGES_REQUESTED** [review]
Need fix Makefile

**Comment by vizhyk** [comment]
@galkinletter  @oleg-h2o It's final version of char. driver. Please, review it.

File: chtest.c [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        return count;							
+}
+
+static void sys_cleanup(void)
+{
+
+	kobject_put(chtest_kobject);
+}
+
+static void __exit chtest_exit(void){
```
Comment (galkinletter) [inline chtest.c]:
```suggestion
static void __exit chtest_exit(void)
{
```
Please fix Kernel code style

File: chtest.c [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+ proc_cleanup();				    //proc_cleanup
+sys_cleanup();       	         		    //sys_cleanup
```
Comment (galkinletter) [inline chtest.c]:
```suggestion
 sys_cleanup();       	         		    //sys_cleanup	
```

Please fix code style also

**Review by galkinletter — CHANGES_REQUESTED** [review]
Please fix code style


check the code formatting

**Comment by vizhyk** [comment]
@galkinletter  @oleg-h2o Fixed code style. Please, check it again.

File: chtest.c [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+	.read = procread,
+	.write = procwrite,
+ };
+		//sys struct
+static struct kobj_attribute sys_attribute =__ATTR(clrflag, 0666, clrflag_show,clrflag_store);
+
+static int __init chtest_init(void)
+ {
+   printk(KERN_INFO "chtest: Initializing the chtest LKM\n");
+if (B_size > 0)
```
Comment (oleg-h2o) [inline chtest.c]: Your indentation below this point is awful. It's almost impossible to read this code.

**Review by oleg-h2o — CHANGES_REQUESTED** [review]
Do read the CodeStyle doc for the kernel and fix indentation.

File: chtest.c [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+{
+
+		kobject_put(chtest_kobject);
+}
+
+
+static ssize_t dev_read(struct file *filep, char *buffer, size_t len, loff_t *offset)
+{
+		int error_count = 0;
+ if(B_size==0){
```
Comment (oleg-h2o) [inline chtest.c]: It make almost impossible to read the file with these indentations.

**Review by oleg-h2o — CHANGES_REQUESTED** [review]
Plz fix indentation ASAP.

**Pull request merged**